pkg/auth/providers/oidc/oidc_client_test.go

```diff
@@ -184,3 +187,83 @@ func TestAddCertKeyToContext(t *testing.T) {
 		})
 	}
 }
+
+func TestFetchAuthURL(t *testing.T) {
+	testCases := []struct {
+		name         string
+		config       map[string]interface{}
+		mockResponse string
+		mockStatus   int
+		expectedURL  string
+		expectError  bool
+	}{
+		{
+			name: "AuthEndpoint already configured",
+			config: map[string]interface{}{
+				"authEndpoint": "https://ranchertest.io/auth",
+			},
+			expectedURL: "https://ranchertest.io/auth",
+			expectError: false,
+		},
+		{
+			name: "Issuer URL provided, valid discovery document",
+			config: map[string]interface{}{
+				"issuer": "https://ranchertest.io",
+			},
+			mockResponse: `{"authorization_endpoint": "https://ranchertest.io/auth"}`,
+			mockStatus:   http.StatusOK,
+			expectedURL:  "https://ranchertest.io/auth",
+			expectError:  false,
+		},
+		{
+			name: "Issuer URL provided, invalid discovery document",
+			config: map[string]interface{}{
+				"issuer": "https://ranchertest.io",
+			},
+			mockResponse: `{"authorization_endpoi": "https://ranchertest.io/auth"}`,
+			mockStatus:   http.StatusOK,
+			expectedURL:  "",
+			expectError:  true,
+		},
+		{
+			name: "Issuer URL provided, error fetching discovery document",
+			config: map[string]interface{}{
+				"issuer": "https://ranchertest.io",
+			},
+			mockResponse: `Internal Server Error`,
+			mockStatus:   http.StatusInternalServerError,
+			expectedURL:  "",
+			expectError:  true,
+		},
+		{
+			name:        "Both authEndpoint and issuerURL missing",
+			config:      map[string]interface{}{},
+			expectedURL: "",
+			expectError: true,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
```

Comment:
Should we run these in parallel? We don't have to though.

```suggestion
		tc := tc
		t.Run(tc.name, func(t *testing.T) {
			t.Parallel()
```

Reply:
sure, doesn't hurt